lib/ddtrace/tracer.rb

```diff
@@ -233,6 +233,8 @@ def start_span(name, options = {})
         if ctx && ctx.trace_id && ctx.span_id
```

Comment:
Curious, why do we have this validation here? I would sort of assume that if we explicitly pass in a context it should just be used, no?

What if the customer is doing their own distributed tracing/tracking and choose to not set a parent id?

I would think if the propagation extractor returns a context we should assume it is valid.

Reply:
This check probably prevented the tracer from crashing; if it unconditionally read from the context, then it would have set the span ID to `nil` and bad things would have happened.

I think you could make the argument that `Context` should be responsible for verifying it has valid values, such that we only have to check `ctx.nil?`. However, adding such validation could cause exceptions elsewhere, if other parts of code are assigning invalid values to `Context`.

Reply:
Actually now that I think about it, it is technically legal that a `Context` has a `nil` trace ID or span ID, particularly if there's no trace active (the default state of a `Context`.) So there's no means of bumping this check to the `Context`.

We could rework this specific check slightly though, such that its more permissive about allowing some values even if not all are available e.g.

```ruby
if ctx
  span.trace_id = ctx.trace_id unless ctx.trace_id.nil?
  span.parent_id = ctx.span_id unless ctx.span_id.nil?
end
```

Doing so would eliminate the explicit synthetics check, and be generally trusting of whatever the `Context` gives it.

Reply:
I applied the above change, which I think resolves this one.